### DISCUSSION TITLE
Why does `poetry show --only main` includes dependencies that are not listed in pyproject.toml?

### DISCUSSION THREAD

**Discussion by roniemartinez** [python-poetry/poetry#6554] [Q&A] [answered]
Hi Poetry team,

I wanted to get a list of licenses for the modules that I am using using `pip-licenses`. 

```
pip-licenses --format=plain-vertical --with-license-file --no-license-path --packages $(poetry show --only main | awk '{print $1}')
```

However, I cannot list my top-level dependencies properly because the output of `poetry show --only main` includes sub-dependencies (or dependencies that are not listed in main group). 

Is there a way to only list the modules that I directly used (EDIT: only the modules listed in `pyproject.toml`)?

**Comment by radoering** [comment, marked as answer]
May I ask why you are only interested in top level dependencies? Normally, if you care for licenses, the license terms of transient dependencies of the main group will be just as important as those of top level dependencies of that group.

**Reply by roniemartinez** [reply]
You are right. But transitive dependencies normally have links from the libraries that directly use them. I believe it should be enough to just show the direct dependencies. Another reason is that since there are several dependencies, the output becomes too bloated (`--format=plain-vertical vertical --with-license-file --no-license-path`).

**Reply by neersighted** [reply]
`poetry show --tree` will get you the information you want but is obviously difficult to parse. You really should not parse `poetry show` anyway -- the output format is unstable. We have a medium-term goal of decoupling the rendering and walking code in `poetry show` so that we can support JSON output, which would be the ideal solution here (e.g. `poetry show --tree --json`).

**Reply by roniemartinez** [reply]
Thanks @neersighted 
While JSON support is not there, I think I can work with this:

```
poetry show --tree --only main | awk '{print $1}' | grep "^[a-zA-Z_]"
```

**Reply by neersighted** [reply]
Just don't be surprised if the output format moves out from under you -- human-readable output of commands could change at any time, even in patch versions.
